### PR TITLE
Add documentation for the teleport system.

### PR DESCRIPTION
**Pull request by wiwei**
https://github.com/microsoft/MixedRealityToolkit-Unity/issues/5282

The teleport system is primarily interesting for VR cases, and I added some docs here mostly so that I could have a section to put the blurb on "how to register for teleport events"

We'd gotten some questions on how to use it (see the issue lined above) and it's clearly not-obvious how you get events, so this is necessary.

### CODE REVIEW

File: Documentation/TeleportSystem/Overview.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+The teleport system is a sub-system of the MRTK that handles teleporting the user when the
+application is using an opaque display. For AR experiences (like HoloLens), the teleportation
+system is not active. For immersive HMD experiences (OpenVR, WMR) the teleport system can
+be enabled.
+
+## Enabling and disabling
+
+The teleport system can be enabled or disabled by toggling the checkbox in its profile:
+
+![Teleport system profile](../../Documentation/Images/TeleportSystem/profile.png)
```
Comment (thalbern) [inline Documentation/TeleportSystem/Overview.md]: do we actually need that image or are we ok with just describing the path of that checkbox / property?

we have a docs guideline for images to only include them if absolutely necessary. Someone would have to update them whenever that profile changes
https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Contributing/DocumentationGuide.html#images--screenshots
(I know we're already showing profile screenshots in a lot of places)

Comment (wiwei) [reply Documentation/TeleportSystem/Overview.md]: That's fair - I think in this case the profile is simple enough that text is sufficient. I've gotten rid of the image.

File: Documentation/TeleportSystem/Overview.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+void EnableTeleportSystem()
+{
+    CoreServices.TeleportSystem.Enable();
+}
+```
+
+## Events
+
+The teleport system exposes events through the [`IMixedRealityTeleportHandler`](xref:Microsoft.MixedReality.Toolkit.Teleport.IMixedRealityTeleportHandler)
```
Comment (Troy-Ferrell) [inline Documentation/TeleportSystem/Overview.md]: Should we link to teleport pointers in the pointers.md?

File: Documentation/TeleportSystem/Overview.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    void OnDisable()
+    {
+        // Unregistering when disabled is important, otherwise this class will continue
+        // to recieve teleportation events.
+        CoreServices.TeleportSystem.UnregisterHandler<IMixedRealityTeleportHandler>(this);
+    }
+
+    void Update()
```
Comment (Troy-Ferrell) [inline Documentation/TeleportSystem/Overview.md]: update seem unnecessary

File: Documentation/TeleportSystem/Overview.md (hunk 2 above)
Comment (wiwei) [reply Documentation/TeleportSystem/Overview.md]: I think there's some value to this, I haven't quite figured out the content yet, so I'll probably proceed with this change first and we can add the other parts as it comes to me.

File: Documentation/TeleportSystem/Overview.md (hunk 3 above)
Comment (wiwei) [reply Documentation/TeleportSystem/Overview.md]: Yep, removed

**Pull request merged**